src/solve.jl

```diff
@@ -22,15 +22,12 @@ function __solve(prob::OptimizationProblem, opt::Optim.AbstractOptimizer;cb = (a
 			x = prob.f.f(θ, prob.p)
 			return x[1]
 		end
-		fg! = let res = DiffResults.GradientResult(prob.x)
-			function (G,θ)
-				if G !== nothing
-					prob.f.grad(res, θ)
-					G .= DiffResults.gradient(res)
-				end
-	
-				return _loss(θ)
+		fg! = function (G,θ)
+			if G !== nothing
+				prob.f.grad(G, θ)
```

Comment:
@ChrisRackauckas we decided to drop the use of `GradientResult` since it doesn't behave well all the time with some backends (FiniteDiff and some previous issues that had to be handled in Zygote).

Reply:
Some form of pre-caching will definitely be necessary. GradientResult should only be used with ForwardDiff, and it should be allocated in the call of the optimizer.